bpfd-operator/controllers/bpfd-agent/kprobe-program.go

```diff
@@ -188,6 +177,23 @@ func (r *KprobeProgramReconciler) reconcileBpfdProgram(ctx context.Context,
 		},
 	}
 
+	return loadRequest
+}
+
+// reconcileBpfdPrograms ONLY reconciles the bpfd state for a single BpfProgram.
+// It does not interact with the k8s API in any way.
+func (r *KprobeProgramReconciler) reconcileBpfdProgram(ctx context.Context,
+	existingBpfPrograms map[string]*gobpfd.ListResponse_ListResult,
+	bytecodeSelector *bpfdiov1alpha1.BytecodeSelector,
+	bpfProgram *bpfdiov1alpha1.BpfProgram,
+	isNodeSelected bool,
+	isBeingDeleted bool,
+	mapOwnerStatus *MapOwnerParamStatus) (bpfdiov1alpha1.BpfProgramConditionType, error) {
+
+	r.Logger.V(1).Info("Existing bpfProgram", "ExistingMaps", bpfProgram.Spec.Maps, "UUID", bpfProgram.UID, "Name", bpfProgram.Name, "CurrentKprobeProgram", r.currentKprobeProgram.Name)
+
+	id := string(bpfProgram.UID)
+
```

Comment:
```suggestion
getAndBuild := func() { 
     		bytecode, err := bpfdagentinternal.GetBytecode(r.Client, bytecodeSelector)
		if err != nil {
			return bpfdiov1alpha1.BpfProgCondBytecodeSelectorError, fmt.Errorf("failed to process bytecode selector: %v", err) }
			
		loadRequest := r.buildKprobeLoadRequest(bytecode, id, bpfProgram, mapOwnerStatus.mapOwnerUuid)
}
```

Something like this is what I meant when I said "just put it in an in-line anonymous and call it when needed."

Reply:
Done.